packages/components/src/styles/components/app-header.scss

```diff
@@ -129,8 +129,14 @@
   }
 }
 
-// prevent interaction with AppHeader when SideNav is open/expanded
-.hds-app-frame:has(.hds-side-nav--is-not-minimized) .hds-app-header--is-mobile {
+// Prevent interaction with AppHeader when SideNav is open/expanded
+
+// Logic: if the AppSideNav is "mobile" (operates in overlay mode) - AND - it is expanded (not minimized),
+// we disable interactions in the AppHeader
```

Comment:
We may want to eventually look to rename or simplify how some of the various responsive classnames get added for better clarity but this update should now correctly target the AppHeader actions in the correct views.